src/main/scala/chiseltest/stage/ChiselTestAnnotations.scala

```diff
@@ -0,0 +1,141 @@
+import firrtl.options.{HasShellOptions, ShellOption, Unserializable}
+
+/** Indicates that this is an [[firrtl.annotations.Annotation Annotation]] directly used in the construction of a
+  * [[ChiselTestOptions]] view.
+  */
+sealed trait ChiselTestOption extends Unserializable {
+  this: Annotation =>
+}
+
+sealed trait ChiselTestOptionObject extends NoTargetAnnotation with HasShellOptions with ChiselTestOption
+
+/* Waveform Annotations.
+ *
+ * @todo Different backends will have different ways of expressing things like WriteVCD,
+ *   there should be a formal mapping from testers2 options form to backend specific forms.
+ *
+ */
+case object WriteVcdAnnotation extends ChiselTestOptionObject {
+  val options: Seq[ShellOption[_]] = Seq(
+    new ShellOption[Unit](
+      longOption = "t-write-vcd",
+      toAnnotationSeq = _ => Seq(WriteVcdAnnotation),
+      helpText = "writes vcd execution log, this option may be moved into firrtl in the future"
+    )
+  )
+}
+/* Coverage Annotations. */
+trait CoverageAnnotations extends NoTargetAnnotation with ChiselTestOptionObject
+
+case object LineCoverageAnnotation extends CoverageAnnotations {
+  val options: Seq[ShellOption[_]] = Seq(
+    new ShellOption[Unit](
+      longOption = "t-line-coverage",
+      toAnnotationSeq = _ => Seq(LineCoverageAnnotation),
+      helpText = "adds line coverage in VCS or Verilator"
+    )
+  )
+}
+
+case object ToggleCoverageAnnotation extends CoverageAnnotations {
+  val options: Seq[ShellOption[_]] = Seq(
+    new ShellOption[Unit](
+      longOption = "t-toggle-coverage",
+      toAnnotationSeq = _ => Seq(ToggleCoverageAnnotation),
+      helpText = "adds toggle coverage in VCS or Verilator"
+    )
+  )
+}
+
+case object BranchCoverageAnnotation extends CoverageAnnotations {
+  val options: Seq[ShellOption[_]] = Seq(
+    new ShellOption[Unit](
+      longOption = "t-branch-coverage",
+      toAnnotationSeq = _ => Seq(ToggleCoverageAnnotation),
+      helpText = "adds branch coverage in VCS"
+    )
+  )
+}
+
+case object ConditionalCoverageAnnotation extends CoverageAnnotations {
+  val options: Seq[ShellOption[_]] = Seq(
+    new ShellOption[Unit](
+      longOption = "t-conditional-coverage",
+      toAnnotationSeq = _ => Seq(ToggleCoverageAnnotation),
+      helpText = "adds conditional coverage in VCS"
+    )
+  )
+}
+
+case object StructuralCoverageAnnotation extends CoverageAnnotations {
+  val options: Seq[ShellOption[_]] = Seq(
+    new ShellOption[Unit](
+      longOption = "t-structural-coverage",
+      toAnnotationSeq = _ => Seq(LineCoverageAnnotation),
+      helpText = "adds all forms of structural in VCS or Verilator"
+    )
+  )
+}
+
+case object UserCoverageAnnotation extends CoverageAnnotations {
+  val options: Seq[ShellOption[_]] = Seq(
+    new ShellOption[Unit](
+      longOption = "t-user-coverage",
+      toAnnotationSeq = _ => Seq(LineCoverageAnnotation),
+      helpText = "adds user coverage in VCS or Verilator"
+    )
+  )
+}
```

Comment:
@ducky64 
Is this (and below) specific to Verilator and VCS? I get that perhaps they're the only backends that support these annotations, but is there any reason these are limited to those? And is there a way for backends that don't support these features to noisily fail if attempted to be invoked with these annotations?

@sequencer
> Is this (and below) specific to Verilator and VCS?

yes, currently only vcs and verilator.

> And is there a way for backends that don't support these features to noisily fail if attempted to be invoked with these annotations?

It will silent ignore these annotations, it won't fail.

But I think  chipsalliance/treadle#263 support this, and we can add it in the later PR.